modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMStructureServiceTest.java

```diff
@@ -499,6 +542,17 @@ protected void setUpPrincipalThreadLocal() throws Exception {
 		PrincipalThreadLocal.setName(_siteAdminUser.getUserId());
 	}
 
+	private DDMStructure _addStructure(String externalReferenceCode)
+		throws Exception {
+
+		DDMStructure ddmStructure = addStructure(
+			group, _classNameId, "Basic Structure");
+
+		ddmStructure.setExternalReferenceCode(externalReferenceCode);
+
+		return DDMStructureTestUtil.updateStructure(ddmStructure);
```

Comment:
Can we use the DDMStructureLocalService instead of using DDMStructureTestUtil?
```
@Inject
private DDMStructureLocalService _ddmStructureLocalService;
```

And you could update:
```
_ddmStructureLocalService.updateStructure(
	structure.getUserId(), structure.getStructureId(),
	structure.getParentStructureId(),
	Collections.singletonMap(LocaleUtil.getSiteDefault(), name),
	structure.getDescriptionMap(), structure.getDDMForm(),
	structure.getDDMFormLayout(),
	ServiceContextTestUtil.getServiceContext(group.getGroupId()));
```

Can you please refactor and test it to see if it will work?

Reply:
I had to use a different updateStructure method that allows for updating ERC, but this suggestion worked.